File: web-service/api-project/src/main/java/io/ehdev/conrad/service/api/service/model/version/GetAllVersionsResponse.java
package io.ehdev.conrad.service.api.service.model.version;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import org.springframework.hateoas.ResourceSupport;

import java.util.ArrayList;
import java.util.List;

public class GetAllVersionsResponse extends ResourceSupport {

    @JsonPropertyDescription("All of the commits for a repository")
    @JsonProperty("commits")
    List<GetAllVersionsCommitResponse> commits = new ArrayList<>();

    @JsonProperty("latest")
    GetAllVersionsCommitResponse latest;

    public List<GetAllVersionsCommitResponse> getCommits() {
        return commits;
    }

    public void addCommits(GetAllVersionsCommitResponse commit) {
        this.commits.add(commit);
    }

    public GetAllVersionsCommitResponse getLatest() {
        return latest;
    }

    public void setLatest(GetAllVersionsCommitResponse latest) {
        this.latest = latest;
    }
}

File: api/rest-model/src/main/kotlin/io/ehdev/conrad/model/version/GetAllVersionsResponse.kt
package io.ehdev.conrad.model.version

import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonPropertyDescription
import io.ehdev.conrad.model.DefaultResourceSupport

class GetAllVersionsResponse() : DefaultResourceSupport() {
    @JsonPropertyDescription("All of the commits for a repository")
    @JsonProperty("commits")
    val commits: MutableList<CommitModel> = mutableListOf()

    @JsonProperty("latest")
    var latest: CommitModel? = null

    class CommitModel(commitId: String, version: String) : BaseVersionResponse(commitId, version)

    fun addCommit(commit: CommitModel) {
        commits.add(commit)
        latest = commit
    }
}
